### PR TITLE
reference/tidb-binlog: add docs about relay log

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? 

Add docs about relay log.

This PR only includes updates that apply to dev, v3.1, and v3.0. 
Updates that do not apply to v3.1 and v3.0 will be made in other PRs.

### What is the related PR or file link(s)? 

- Reference link(s):
- This PR is to align with: https://github.com/pingcap/docs-cn/pull/2224
- [ ] N/A (not applicable)

**Important Notice:**

**If your changes apply to multiple TiDB documentation versions**, to trigger the bot to cherry-pick this PR to other release branches, make sure you add labels such as:

- **needs-cherry-pick-3.0**
- **needs-cherry-pick-3.1**
- **needs-cherry-pick-2.1**

### CODE REVIEW

**Comment by ran-huang** [comment]
A minor update from https://github.com/pingcap/docs-cn/pull/2308 is included in the latest commit.

File: reference/tidb-binlog/deploy.md [hunk 1]
```diff
@@ -551,6 +551,13 @@ The following part shows how to use Pump and Drainer based on the nodes above.
 
         # replicate-do-db = ["~^b.*","s1"]
 
+        # [syncer.relay]
+        # It saves the catalog of the relay log. The relay log is not enabled if the value is empty.
```
Comment (lilin90) [inline reference/tidb-binlog/deploy.md]:
```suggestion
        # It saves the directory of the relay log. The relay log is not enabled if the value is empty.
```

File: reference/tidb-binlog/relay-log.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Consistent state during Drainer replication
+
+The downstream clusters reaching a consistent state means the data of the downstream clusters are the same as the snapshot of the upstream which sets `tidb_snapshot = ts`.
+
+The checkpoint consistency means Drainer checkpoint saves the consistent state of replication in `consistent`. When Drainer runs, `consistent` is `false`. After Drainer exits normally, `consistent` is set to `true`.
+
+You can query the downstream checkpoint table as follows:
+
+```
+mysql> select  * from tidb_binlog.checkpoint;
```
[the remark this answers is not in the document]
Comment (ran-huang) [reply reference/tidb-binlog/relay-log.md]: @lilin90 Shall I also remove `mysql>` from the Chinese version?

**Comment by ran-huang** [comment]
@lilin90 PTAL😀

File: reference/tidb-binlog/relay-log.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+---
+title: TiDB Binlog Relay Log
+category: reference
```
Comment (lilin90) [inline reference/tidb-binlog/relay-log.md]: Please add a one-sentence summary here.
```suggestion
summary: xxx
category: reference
```

**Comment by lilin90** [comment]
For the code block format, see https://github.com/pingcap/docs/pull/1881/commits/678bdbbf62a3617c00f39ba8c4775f6ac6827e6c. This is to make the SQL statement line friendly for users to copy it.

**Comment by ran-huang** [comment]
@lilin90 PTAL~

**Comment by ran-huang** [comment]
@july2993 PTAL~

**Pull request merged**